### PR TITLE
Added custom naming support

### PR DESCRIPTION
**Pull request by AntonBazhal**
## What did you implement:

Sometimes it's needed to control naming for alarms. Some use cases are:
1. Need to mark important alarms with keyword
1. Make alarm names compliant with company standards
1. Make alarm names more human-readable, especially when using nested stacks

## How did you implement it:

Added `nameTemplate` property to config (root level and alarm level) that supports interpolation.

## How can we verify it:

Use template like `$[functionName]-$[metricName]-Alarm`, and check that CF resource has `AlarmName` property formatted according to your template.

## Todos:

- [x] Write tests
- [x] Write documentation
- [x] Fix linting errors
- [x] Provide verification config/commands/resources

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -231,11 +243,11 @@ class Plugin {
       const normalizedFunctionName = this.providerNaming.getLambdaLogicalId(functionName);
 
       const functionAlarms = this.getFunctionAlarms(functionObj, config, definitions);
-      const alarms = globalAlarms.concat(functionAlarms);
+      const alarms = globalAlarms.concat(functionAlarms).map(alarm => _.defaults({ nameTemplate: config.nameTemplate }, alarm));
```
Comment (AntonBazhal) [inline src/index.js]: Makes overwriting possible

**Comment by AntonBazhal** [comment]
@johncmckim

**Comment by AntonBazhal** [comment]
@johncmckim

**Comment by rschick** [comment]
@AntonBazhal any movement on this?

**Comment by AntonBazhal** [comment]
@rschick this is merged into our fork

**Comment by Limess** [comment]
This is what we basically wanted when writing https://github.com/ACloudGuru/serverless-plugin-aws-alerts/pull/89, an escape hatch seemed sensible at the time given the state of the project, but this would also be nice to have as first class support for custom names.

**Comment by bulkan** [comment]
@AntonBazhal sorry it's taking a while to get back to you. Can you please update the PR ?

**Comment by AntonBazhal** [comment]
Updated in PR #97

**Pull request closed without merging**